xbmc/filesystem/File.cpp

```diff
@@ -283,7 +288,10 @@ bool CFile::Open(const CURL& file, const unsigned int flags)
         return m_pFile->Open(url);
       }
     }
-    m_pFile = CFileFactory::CreateLoader(url);
+    if (m_pFile)
+      m_pFile->Close();
+    else
+      m_pFile = CFileFactory::CreateLoader(url);
```

Comment:
keep-alive (reusing of conection) works only on the same curl_easy handle.
You can call multiple times open() now, was not possible before.

(yes it was possible, but lead to memory leaks :-) So its more an bugfix as an feature).